contracts/oracle/collateralization/CollateralizationOracleGuardian.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./ICollateralizationOracleWrapper.sol";
+import "../../refs/CoreRef.sol";
+import "../../utils/Timed.sol";
+import "../../Constants.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/** 
+    @title Fei Protocol's Collateralization Oracle Guardian
+    @author Fei Protocol
+    This contract needs to be granted the ORACLE_ADMIN role
+    The guardian can leverage this contract to make small bounded changes to CR
+    This is intended to be used only in emergencies when the base CollateralizationOracle is compromised
+    The guardian should be able to approximate manual changes to CR via this contract without retaining too much power
+*/
+contract CollateralizationOracleGuardian is CoreRef, Timed {
+    using SafeCast for uint256;
+
+    event DeviationThresholdUpdate(uint256 oldDeviationThresholdBasisPoints, uint256 newDeviationThresholdBasisPoints);
+
+    /// @notice the oracle wrapper to update
+    ICollateralizationOracleWrapper public immutable oracleWrapper;
+
+    /// @notice the maximum update size relative to current, measured in basis points (1/10000)
+    uint256 public deviationThresholdBasisPoints;
+    
+    /**
+        @notice The constructor for CollateralizationOracleGuardian
+        @param _core the core address to reference
+        @param _oracleWrapper the instance of CollateralizationOracleWrapper
+        @param _frequency the maximum frequency a guardian can update the cache
+        @param _deviationThresholdBasisPoints the maximum percent change in a cache value for a given update
+     */
+    constructor(
+        address _core, 
+        ICollateralizationOracleWrapper _oracleWrapper, 
+        uint256 _frequency,
+        uint256 _deviationThresholdBasisPoints
+    ) CoreRef(_core) Timed(_frequency) {
+        oracleWrapper = _oracleWrapper;
+
+        _setDeviationThresholdBasisPoints(_deviationThresholdBasisPoints);
+
+        _initTimed();
+    }
+
+    /// @notice guardian set the cache values on collateralization oracle
+    /// @param protocolControlledValue new PCV value
+    /// @param userCirculatingFei new user FEI value
+    /// @dev make sure to pause the CR oracle wrapper or else the set value would be overwritten on next update
```

Comment:
Why not add logic in the collateralization oracle and this contract so that if you call `setCache`, it automatically pauses the collateralization oracle if it is not paused yet.